pennylane/pauli/utils.py

```diff
@@ -1145,7 +1152,9 @@ def diagonalize_pauli_word(pauli_word):
     return diag_term
 
 
-def diagonalize_qwc_pauli_words(qwc_grouping):  # pylint: disable=too-many-branches
+def diagonalize_qwc_pauli_words(
+    qwc_grouping,
+):  # pylint: disable=too-many-branches, isinstance-second-argument-not-valid-type
```

Comment:
Why is this needed? Pylint doesn't raise any error for me.

Reply:
And all types used in the added isinstance checks should be valid...

Reply:
Yes, but I think the issue is the tuple. For some reason pylint thinks we are comparing the op to the class `tuple` even though the function accepts an iterable in the second argument.

It is described [here](https://github.com/PyCQA/pylint/issues/3507), I think we need to upgrade pylint.

Reply:
Where was the error raised, in your local `pre-commit` or in the github action?

If I remember correctly, I used several times a `tuple` in the `isinstance` check and I never got a complain from pylint...

Reply:
In GitHub actions, I can undo the change and we can double check.